### PR TITLE
Neural SDE tutorial

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Based off of the old one. Might be easiest to just make this Flux based?

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@Abhishek-1Bhatt you're finishing this one? I think this is the last one to finish DiffEqFlux (for now).

File: docs/src/examples/neural_sde.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                       Lux.Dense(50, 2))
+p1, st1 = Lux.setup(rng, drift_dudt)
+
+diffusion_dudt = Lux.Chain(Lux.Dense(2, 2))
+p2, st2 = Lux.setup(rng, diffusion_dudt)
+
+p1 = Lux.ComponentArray(p1)
+p2 = Lux.ComponentArray(p2)
+#Component Arrays doesn't provide a name to the first ComponentVector, only subsequent ones get a name for dereferencing
+p = [p1, p2]
```
Comment (avik-pal) [inline docs/src/examples/neural_sde.md:87]: I will write a manual section later tonight in Lux itself, but this not quite the right interface. The "correct" thing to do would be 
```suggestion
p = Lux.ComponentArray((p1 = p1, p2 = p2))
```
Then index the parameters with `.p1` and `.p2`. This makes it consistent with the Lux ContainerLayer Interface.

The same thing holds for the states

Comment (Abhishek-1Bhatt) [reply docs/src/examples/neural_sde.md:87]: Would it be alright to make the user follow a strict naming scheme of p1 and p2, just a small concern.

Comment (avik-pal) [reply docs/src/examples/neural_sde.md:87]: So the user should not be responsible for constructing this in general. Lux.setup on NeuralDSDE should return the parameters in this format

Comment (avik-pal) [reply docs/src/examples/neural_sde.md:87]: The convention I follow in Lux is to document what the parameter and state names are http://lux.csail.mit.edu/stable/devdocs/style_guide/#documentation-for-layers, and construct the structure in `Lux.setup`. After that, if the user messes up then nothing can be done but redirect them to the documentation

Comment (Abhishek-1Bhatt) [reply docs/src/examples/neural_sde.md:87]: Okay, lets finalize this for all the Lux constructors in DiffEqFlux then

Comment (ChrisRackauckas) [reply docs/src/examples/neural_sde.md:87]: Yes, the user shouldn't construct this in general. This is just a start from a copy paste of broken code so don't take it as a correct way to do things for this. As mentioned in https://github.com/SciML/DiffEqFlux.jl/issues/727, these should be an `AbstractExplicitLayer`, which means they should do things exactly like `Dense`. They should have one state, take in a state, and return a state. They should take in a neural network definition and give you back a state from `setup`. Basically, it should act exactly like Dense does, and be able to perfectly swap in without any other code changes, and if not it's wrong. The only thing that should be different is the constructor for the layer.

**Comment by Abhishek-1Bhatt** [comment]
We can merge #738 for now as discussed in https://github.com/SciML/DiffEqFlux.jl/pull/736#issuecomment-1163634256 and switch back to Lux with minor edits as soon as the issues are resolved.

**Pull request merged**